src/implementations/twig/components/accordion/accordion.story.js

```diff
@@ -1,40 +1,58 @@
-import { withKnobs, text, select, optionsKnob } from '@storybook/addon-knobs';
 import { withNotes } from '@ecl/storybook-addon-notes';
-import { getExtraKnobs, tabLabels } from '@ecl/story-utils';
 import withCode from '@ecl/storybook-addon-code';
 import defaultSprite from '@ecl/resources-ec-icons/dist/sprites/icons.svg';
 import demoData from '@ecl/specs-component-accordion/demo/data';
 
 import accordion from './accordion.html.twig';
 import notes from './README.md';
 
-const prepareAccordion = (data) => {
-  data.items.forEach((item, index) => {
-    const levels = [1, 2, 3, 4, 5, 6];
-    const { id, level, toggle, content } = item;
-    item.id = text(`items[${index}].id`, id, tabLabels.required);
-    item.content = text(`items[${index}].content`, content, tabLabels.required);
-    item.toggle.label = text(
-      `items[${index}].toggle.label`,
-      toggle.label,
-      tabLabels.required
-    );
-    item.toggle.icon.path = optionsKnob(
-      `items[${index}].toggle.icon.path`,
-      { current: defaultSprite, 'no path': '' },
-      defaultSprite,
-      { display: 'inline-radio' },
-      tabLabels.required
-    );
-    item.level = select(
-      `items[${index}].level`,
-      levels,
-      level,
-      tabLabels.optional
-    );
-  });
+const argTypes = {};
+demoData.items.forEach((item, i) => {
+  argTypes[`toggle${i + 1}`] = {
+    name: `toggle ${i + 1}`,
+    type: { name: 'string', required: true },
+    defaultValue: item.toggle.label,
+    description: `Text of the accordion toggler`,
+    table: {
+      type: { summary: 'string' },
+      defaultValue: { summary: '' },
+      category: 'Togglers',
+    },
+    control: {
+      type: 'text',
+    },
+  };
+  argTypes[`content${i + 1}`] = {
+    name: `content ${i + 1}`,
+    type: { name: 'string', required: true },
+    defaultValue: item.content,
+    description: 'Text of the hidden content',
+    table: {
+      type: { summary: 'string' },
+      defaultValue: { summary: '' },
+      category: 'Hidden content',
+    },
+    control: {
+      type: 'text',
+    },
+  };
+});
 
-  getExtraKnobs(data);
+const prepareDataWithArgs = (data, args) => {
```

Comment:
I am not sure whether the change in name of this function matches what it actually does.
From the pull request for controls in the button component I put a comment about it. If we rename the function, which i think we should do, it should be related to specs, not args.
We might need to align whether this function will be the place where defaults are applied. 
At any case, it will be useful to have consistent naming and operation in this function throughout all stories.